src/jaxsim/rbda/contacts/rigid.py

```diff
@@ -0,0 +1,504 @@
+
+            inactive = pz > terrain_height
+
+            # Compute the terrain normal and the contact depth
+            n̂ = terrain_normal.squeeze()
+            h = jnp.array([0, 0, terrain_height - pz])
+
+            # Compute the penetration depth normal to the terrain.
+            δ = jnp.maximum(0.0, jnp.dot(h, n̂))
+
+            # Compute the penetration normal velocity.
+            δ̇ = -jnp.dot(W_o_dot_C, n̂)
+
+            return inactive, (δ, δ̇)
+
+        inactive_collidable_points, (delta, delta_dot) = jax.vmap(detect_contact)(
+            W_o_C, W_o_dot_C, terrain_height, terrain_normal
+        )
+
+        return inactive_collidable_points, (delta, delta_dot)
+
+    @staticmethod
+    def _delassus_matrix(
+        M: jtp.Matrix,
+        J_WC: jtp.Matrix,
+    ):
+        sl = jnp.s_[:, 0:3, :]
+        J_WC_lin = jnp.vstack(J_WC[sl])
+
+        delassus_matrix = J_WC_lin @ pinv(M) @ J_WC_lin.T
+        return delassus_matrix
+
+    @staticmethod
+    def _compute_ineq_constraint_matrix(
+        inactive_collidable_points: jtp.Vector, mu: jtp.Float
+    ) -> jtp.Matrix:
+        def compute_G_single_point(mu: float, c: float) -> jtp.Matrix:
+            """
+            Compute the inequality constraint matrix for a single collidable point
+            Rows 0-3: enforce the friction pyramid constraint,
+            Row 4: last one is for the non negativity of the vertical force
+            Row 5: contact complementarity condition
+            """
+            G_single_point = jnp.array(
+                [
+                    [1, 0, -mu],
+                    [0, 1, -mu],
+                    [-1, 0, -mu],
+                    [0, -1, -mu],
+                    [0, 0, -1],
+                    [0, 0, c],
+                ]
+            )
+            return G_single_point
+
+        G = jax.vmap(compute_G_single_point, in_axes=(None, 0))(
+            mu, inactive_collidable_points
+        )
+        G = block_diag(*G)
+        return G
+
+    @staticmethod
+    def _compute_ineq_bounds(n_collidable_points: jtp.Float) -> jtp.Vector:
+        n_constraints = 6 * n_collidable_points
+        return jnp.zeros(shape=(n_constraints,))
+
+    @staticmethod
+    def _compute_mixed_nu_dot_free(
+        model: js.model.JaxSimModel,
+        data: js.data.JaxSimModelData,
+        references: js.references.JaxSimModelReferences | None = None,
+    ) -> jtp.Array:
+        references = (
+            references
+            if references is not None
+            else js.references.JaxSimModelReferences.zero(
+                model=model, data=data, velocity_representation=VelRepr.Mixed
+            )
+        )
+
+        with (
+            data.switch_velocity_representation(VelRepr.Mixed),
+            references.switch_velocity_representation(VelRepr.Mixed),
+        ):
+            BW_v_WB = data.base_velocity()
+            W_o_dot_B = BW_v_WB[0:3]
+            W_omega_WB = BW_v_WB[3:6]
```

Comment:
```suggestion
            W_ṗ_dot_B, W_ω_WB = jnp.split(BW_v_WB, 2)
```

Reply:
Done in e1e0c6c9eea88e9d13b395a0a1e79bc73f620c0e